### PR TITLE
Part 1a is completed

### PR DESCRIPTION
**Pull request by marcandrelabelle**
Part 1a is finished, waiting for corrections.

### CODE REVIEW

**Review by matkatmusic — APPROVED** [review]

File: main.cpp [hunk 1]
```diff
@@ -20,67 +20,96 @@ You are going to write 10 UDTs in project3.
...
+    1) digital signal processing
+    2) imitate tube amp  
+    3) amplifie sound 
+ */
+
+/*
+Thing 3) Sampling Pad
+5 properties:
+    1) number of pads
+    2) set click tempo bpm
```
Comment (matkatmusic) [inline main.cpp]: `set` is an action.  you have defined an action instead of a property.
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -20,67 +20,96 @@ You are going to write 10 UDTs in project3.
...
+    3) set velocity for triggering
```
Comment (matkatmusic) [inline main.cpp]: you have defined an action instead of a property.
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 1 above)
Comment (marcandrelabelle) [reply main.cpp]: I fixed the issues

**Review by matkatmusic — APPROVED** [review]

File: main.cpp [hunk 3]
```diff
@@ -108,7 +107,19 @@ Thing 4) analog console
...
 
-
+/*
+Thing 10) Digital Audio Workstation (DAW)
+5 properties:
+    1) number of maximum tracks
+    2) audio plugin inserts
+    3) bus channels
+    4) editing tools
+    5) loopers
```
Comment (matkatmusic) [inline main.cpp]: All of these can be represented with primitives
```
These parts will not be defined using Primitives, but instead will be their own UDTs you'll define in Part 1d
```
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

**Review by matkatmusic — COMMENTED** [review]
ok good.  Added the next set of instructions.

File: main.cpp [hunk 4]
```diff
@@ -107,6 +128,76 @@ Thing 4) analog console
...
  */
 
+/*
+Thing 5) Mixer
+5 properties:
+    1) faders 
+    2) routing panel 
+    3) record ready btn
+    4) input fx
+    5) input selector
```
Comment (matkatmusic) [inline main.cpp]: These are objects, not primitives!   You need to pick things that can be represented with `int float double bool char std::string`
- [ ] revise

File: main.cpp [hunk 5]
```diff
@@ -107,6 +128,76 @@ Thing 4) analog console
...
+ */
+
+/*
+Thing 7) Toolbar Docker
+5 properties:
+    1) number of tools in docker
+    2) tool selector size  
+    3) docker position 
+    4) docker size
+    5) hide docker btn
```
Comment (matkatmusic) [inline main.cpp]: 5): This is an object, not a primitive!   You need to pick things that can be represented with `int float double bool char std::string`
- [ ] revise

File: main.cpp [hunk 6]
```diff
@@ -107,6 +128,76 @@ Thing 4) analog console
...
+ */
+
+/*
+Thing 8) Performance Meter
+5 properties:
+    1) text size 
+    2) window size
+    3) font style
+    4) background colour
+    5) CPU graph
```
Comment (matkatmusic) [inline main.cpp]: 5) is an object, not a primitive
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
You've got some objects where you need primitives.

**Review by matkatmusic — APPROVED** [review]

File: main.cpp [hunk 7]
```diff
@@ -20,67 +20,483 @@ You are going to write 10 UDTs in project3.
...
+    std::string tempoSignature = "7/8";
+    //enveloppe line colour
+    std::string envLineColour = "purple";
+
+    //change tempo bpm
+    void changeBpm ( int newBpm);
+    //create new point
+    void newPoint ( int newPointPosX, int newPointPosY, bool pointShap); 
+    //reduce number of points
+
```
Comment (matkatmusic) [inline main.cpp]: You're missing your 3rd function
- [ ] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
you forgot one.

File: main.cpp [hunk 8]
```diff
@@ -20,67 +20,484 @@ You are going to write 10 UDTs in project3.
...
+    int samplingSpaceSec = 120;
+    //number of integrated effects
+    int numIntegratedFx = 12;
+
+    //trigger  .wav sounds
+    void trigSnd (int padIndex, bool trig);
+    //play a sequence
+    void playSequence (int numSteps, int bpm, bool startSequence);
+    //send midi
+
```
Comment (matkatmusic) [inline main.cpp]: Where's your 3rd function?
- [x] revise

File: main.cpp [hunk 9]
```diff
@@ -20,67 +20,484 @@ You are going to write 10 UDTs in project3.
...
+        int tapTempoBpm (int numTap, float time);
+        void moveCursorPos (int beat, int mesure);
+    };
+
+    //loop audio and midi sections
+    void loopAudioSection ( float loopIn, float loopOut);
+    //record audio and midi performance 
+    void recordData (int input, bool isMidi);
+    //mix multiple recordings and tracks into a final produced
+    void bounce  ( int tracks, float startingPoint, float endingPoint);
```
Comment (matkatmusic) [inline main.cpp]: Why do you have so much space around your `(`? 
- [x] revise

File: main.cpp [hunk 10]
```diff
@@ -20,67 +20,484 @@ You are going to write 10 UDTs in project3.
...
+    float dockerWidth = 500.0f;
+    //number of row
+    int numDockerRow = 2;
+
+    //change tool for editing
+    void changeTool (int toolListIndex, int replaceIndex);
+    //customize displayed tools
+    void customizeTool (int indexSwitch, int newToolIndex);
+    //activate metronom
+    void activateMetronom  (bool mentroPlay);
```
Comment (matkatmusic) [inline main.cpp]: Why do you have so much space before the `(`?
- [x] revise

File: main.cpp [hunk 11]
```diff
@@ -20,67 +20,484 @@ You are going to write 10 UDTs in project3.
...
+    int numStereoChanels = 2;
+    //pre or post EQ selector
+    bool preFader = false;
+
+    //Mix different sounds together
+    void mixSounds (int trackNum, float trackLvl, float pan); 
+    //contrôl master output level
+    void masterOut (float masterOutdB);
+    //add tone
+    void addTone (float hiFreq, float hiBoost, float midFreq, float midBoost,float bassFreq, float bassBoost);
```
Comment (matkatmusic) [inline main.cpp]: be consistent with the spacing around your `,` 
- [x] revise

File: main.cpp [hunk 12]
```diff
@@ -20,67 +20,484 @@ You are going to write 10 UDTs in project3.
...
+    //fx level
+    float fxvl = 4.3f;
+
+
+    //digital signal processing
+    void signalProcessing (bool processSignal, int sampleStart, int sampleEnd);
+    //imitate tube amp 
+    void imitateTubeAmp (int ampSelectIndex, std::string speakerIr); 
+    //amplifie sound
+    void amplifieSound ( int inputSig, float gain);
```
Comment (matkatmusic) [inline main.cpp]: why do you have so much space around the `(`?
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]
the issues are mainly with matching the course coding style.

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**